functions/utils/index.js

```diff
@@ -0,0 +1,116 @@
+import { RelationKind } from './constants';
+
+const isRecord = (value) =>
+  value &&
+  typeof value === 'object' &&
+  !Array.isArray(value) &&
+  Object.keys(value).length > 0 &&
+  value.id !== undefined;
+
+const isCollection = (value) => Array.isArray(value) && isRecord(value[0]);
+
+const parseBelongsTo = (name, value) => {
+  if (isRecord(value)) {
+    const keys = Object.keys(value);
+    return `${name} {
+      ${keys.map((key) => key).join('\n')}
+    }`;
+  }
+
+  return `${name} {
+    id\n
+  }`;
+};
+
+const parseHasManyAndHasAndBelongsToMany = (name, value) => {
+  if (isCollection(value)) {
+    const keys = Object.keys(value[0]);
+    return `${name} {
+      ${keys.map((key) => key).join('\n')}
+    }`;
+  }
+
+  return `${name} {
+    id\n
+  }`;
+};
+
+const getQueryKeys = (properties) =>
+  properties
+    .map((property) => {
+      const {
+        key: [{ kind, name }],
+        value,
+      } = property;
+
+      switch (kind) {
+        case RelationKind.BELONGS_TO:
+          return parseBelongsTo(name, value);
+        case RelationKind.HAS_MANY:
+        case RelationKind.HAS_AND_BELONGS_TO_MANY:
+          return parseHasManyAndHasAndBelongsToMany(name, value);
+
+        default:
+          return name;
+      }
+    })
+    .join('\n');
+
+const belongsToValue = (value) => (isRecord(value) ? value.id : value);
+
+const hasManyOrHasAndBelongsToManyValue = (value) => {
+  const recordIds = value.map((val) => (isRecord(val) ? val.id : val));
+  return { id: recordIds };
+};
+
+const getAssignedValue = (kind, value) => {
+  switch (kind) {
+    case RelationKind.BELONGS_TO:
+      return belongsToValue(value);
+    case RelationKind.HAS_MANY:
```

Comment:
This is meant to do nothing?

Reply:
This is what I have reused from the native functions repo: https://github.com/bettyblocks/native-functions/blob/master/functions/utils/index.js. There is no need to reinvent the wheel. This notation means however that multiple cases return the same output.